squad-server/index.js

```diff
@@ -254,6 +255,10 @@ export default class SquadServer extends EventEmitter {
       delete data.attackerName;
 
       this.emit('PLAYER_DIED', data);
+
+      if (data.teamkill) {
+        this.emit('TEAMKILL', data);
```

Comment:
Using the "TEAMKILL" event name is problematic because it also occurs when someone is wounded (iirc), thus this change could trigger somewhat duplicate events - not great considering people use this event within plugins that automatically ban players after x teamkill events.

Reply:
According to my testing, the following events exist:
- PLAYER_DAMAGED: the player is hurt but not down
- PLAYER_WOUNDED: the player is downed but revivable
- PLAYER_DIED: the player cannot be revived and must respawn

PLAYER_WOUNDED and PLAYER_DAMAGED are mutually exclusive and preceded by the PLAYER_DAMAGED event.

Currently, only PLAYER_WOUNDED causes a TEAMKILL event and PLAYER_DIED does not.

### Examples

#### Multiple damage events leading to a TK, player is revivable
```log
[2021.10.31-01.13.44:364][426]LogSquad: Player:Watson ActualDamage=26.040001 from Charlz caused by BP_M4_Classic_M68_C_2147477285
[2021.10.31-01.13.45:125][464]LogSquad: Player:Watson ActualDamage=62.000004 from Charlz caused by BP_M4_Classic_M68_C_2147477285
[2021.10.31-01.13.45:125][464]LogSquadTrace: [DedicatedServer]ASQSoldier::Wound(): Player:Watson KillingDamage=62.000004 from BP_PlayerController_C_2147478566 caused by BP_M4_Classic_M68_C_2147477285
```

#### TK victim dies
Caused by instakills and greyed-out kills
```log
[2021.10.31-00.38.35:319][ 83]LogSquad: Player:Victim ActualDamage=900.000000 from Attacker caused by BP_BTR82A_RUS_2A72_AP_C_2147479210
[2021.10.31-00.38.35:319][ 83]LogSquadTrace: [DedicatedServer]ASQSoldier::Die(): Player:Victim KillingDamage=900.000000 from BP_PlayerController_C_2147481608 caused by BP_BTR82A_RUS_2A72_AP_C_2147479210
```

#### Victim is in vehicle
```log
[2021.10.31-00.11.36:390][263]LogSquadTrace: [DedicatedServer]ASQVehicleSeat::TraceAndMessageClient(): SQVehicleSeat::TakeDamage[PointDamage] BP_US_Util_Woodland_Logi_C_2147481948 for 15.300000 damage (type=BP_Kinetic_DamageType_C)
[2021.10.31-00.11.36:390][263]LogSquadTrace: [DedicatedServer]ASQVehicleSeat::TraceAndMessageClient(): Waylander: 15.30 damage taken by causer BP_50Cal_M1151_Woodland_C_2147481916 instigator Watson health remaining 693.23
[2021.10.31-00.11.36:390][263]LogSquadTrace: [DedicatedServer]ASQVehicleSeat::TraceAndMessageClient(): SQVehicleSeat::TakeDamage[PointDamage] BP_US_Util_Woodland_Logi_C_2147481948 for 13.300000 damage (type=BP_Kinetic_DamageType_C)
[2021.10.31-00.11.36:390][263]LogSquadTrace: [DedicatedServer]ASQVehicleSeat::TraceAndMessageClient(): Waylander: 13.30 damage taken by causer BP_50Cal_M1151_Woodland_C_2147481916 instigator Watson health remaining 679.93
[2021.10.31-00.11.36:391][263]LogSquad: Player:nullptr ActualDamage=51.500000 from Watson caused by BP_50Cal_M1151_Woodland_C_2147481916
[2021.10.31-00.11.36:391][263]LogSquadTrace: [DedicatedServer]ASQSoldier::Wound(): Player:nullptr KillingDamage=51.500000 from BP_PlayerController_C_2147481344 caused by BP_50Cal_M1151_Woodland_C_2147481916
```

Reply:
Do you mean PLAYER_WOUNDED and PLAYER_DIED are mutually exclusive? Looks like a typo?

Reply:
> Do you mean PLAYER_WOUNDED and PLAYER_DIED are mutually exclusive? Looks like a typo?

oops, yes, exactly

Reply:
They're not mutually exclusive, both can be trigger of a player is wounded and then gives us.

Reply:
Just tested it and giving up/bleeding out triggers PLAYER_DIED indeed. 

I believe we can fix this by checking whether the damage event has the same chainID as the PLAYER_DIED event. what do you think?

Reply:
That's a possible solution, but not sure how much of a fan I am of the `TEAMKILL` event possibly being triggered at two different times when a player is wounded or killed. I think there need to be two separate events, e.g. `WOUNDED_TEAMKILL` and `DIED_TEAMKILL`, or just not use the events entirely and listen to the `PLAYER_WOUNDED` and `PLAYER_DIED` events and look at the teamkill flag in that.